### ISSUE TITLE
Block switcher menu accessibility and labeling improvements

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#62931]
### Description

This came into my mind while providing a11y feedback on https://github.com/WordPress/gutenberg/issues/60810

The block switcher opens a dropdown menu that is an ARIA menu pattern. ARIA menus can only contain children with role `menuitem`, `menuitemcheckbox`, `menuitemradio` and, optionally, groups with separators.

Visually, the menu is split into two main groups. Semantically, there are two main groups as well. See attached screenshot to compare the visuals with the accessibility tree.

![Screenshot 2024-06-27 at 16 34 07](https://github.com/WordPress/gutenberg/assets/1682452/3dc3c4f8-3a64-46b7-9df9-a11498ccfa39)

However, while the first group is labeled 'TRANSFORM TO`, the second group is unlabeled.

As a user, I'm not sure I would understand why there's a visual separation in two groups. What the difference between the two groups is?

Semantically, it doesn't make much sense labeling only the first group. Grouping items together makes only sense if the scope and purpose of the group is clear and meaningful. Is there a good reason to have two main groups in the first place? If there is one, than both groups should be visually and semantically labeled,

Note: the third group with the 'this block is connected' information should be removed as pointed out on https://github.com/WordPress/gutenberg/issues/60810#issuecomment-2194908828

### Step-by-step reproduction instructions

- Select a block.
- Open the block switcher from the block toolbar.
- Note: if the block is connected to external sources via the Bindings API, it will show a third group at the bottom. This is not much relevant for this issue.
- Observe the menu is visually split in two groups;
- Observe the second group is not labeled.
- Optionally, inspect the accessibility tree in your browser dev tools and observe the secodn group is unlabeled.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by sarthaknagoshe2002** [comment]
@afercia 
To improve clarity and accessibility, I recommend labeling the second group in the block switcher dropdown as "Advanced Transformations", complementing the existing "Transform To" label for the first group.

Maintaining this separation is necessary as the groups serve distinct purposes: the first focuses on basic transformations, while the second offers more advanced or specialized options.

Let me know your thoughts!

**Comment by afercia** [comment]
@sarthaknagoshe2002 thanks, yes I'd totally agree the second group needs a visible title to be properly labeled. This issue is still waiting for some design feedback though. Cc @WordPress/gutenberg-design

**Comment by sarthaknagoshe2002** [comment]
Umm I feel we can at least add the label as a temporary fix until we get the design feedbacks for this from the design team.

**Comment by sarthaknagoshe2002** [comment]
Also, I was not able to reproduce this

> Note: the third group with the 'this block is connected' information should be removed as pointed out on https://github.com/WordPress/gutenberg/issues/60810#issuecomment-2194908828

**Comment by jameskoster** [comment]
@afercia did you see the latest designs inhttps://github.com/WordPress/gutenberg/issues/40208, which propose utilising [`Menu`](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-menu--with-submenu) for this UI? Hopefully this would address most of the a11y issues?

**Comment by afercia** [comment]
@jameskoster thanks, will comment there.

**Comment by sarthaknagoshe2002** [comment]
@afercia Do you think that we should move ahead with the changes I suggested as a temporary fix until the new design is implemented?

**Comment by afercia** [comment]
@sarthaknagoshe2002 personally I'd like to move ahead with your PR. As I [commented](https://github.com/WordPress/gutenberg/issues/40208#issuecomment-2507456867) on the issue with the new design, I'm not sure the new designis any read as it mixes together differenc concept sin the same menu and it should be reconsidered IMHO.

**Comment by sarthaknagoshe2002** [comment]
@afercia Then lets get the PR merged.

**Comment by afercia** [comment]
@sarthaknagoshe2002 I'm changing my mind. The more I think at it, the more I lean towards showing all transforms within the same group, with no separation. Thanks for your patience 🙏🏻 


Looking a bit more into the logic of this menu, so far it shows some 'priority' transforms that are shown at the top only for some text blocks. When there are both priority transforms and other transforms, the 'rest' of the transforms is shown in a separate group.

While I agree to show the priority transforms at the top, I'm not sure it makes sense to visually and semantically separate the two groups. Why they should be separated in the first place? I would argue that this menu may show other transforms that are conceptually different from ordinary transforms, for example: the block variations (e.g. for a Group block) are conceptually different from ordinary transforms, still they are shown together with all the transforms:

In the screenshot below, from the left, the menus of the blocks: Paragraph, Image, Group, Quote.

![Image](https://github.com/user-attachments/assets/3db7a52b-54fa-4212-9533-5f1befa83c5f)

Basically I'm not sure why we should show a separation for the priority transforms and not for other logically grouped transforms like the cariations. 

At this point I'd tend to think it would be best to simplify and render only one`MenuGroup` with no separation.
Cc @jameskoster any objection to removing the separation?

The current PR at https://github.com/WordPress/gutenberg/pull/67397 adds a label to the second group but I'm not sure it would be ideal in some cases. Screenshot:

![Image](https://github.com/user-attachments/assets/7d82cad9-0e87-4144-92da-c1a668bc18a1)

**Comment by jameskoster** [comment]
I tend to agree that the separation doesn't really add much, if anything it creates confusion by suggesting the transforms are different in some way. It's worse when Styles also appear. The most 'helpful' transforms are listed first, and that seems adequate to me.

 I'd welcome more feedback from @WordPress/gutenberg-design though.

**Comment by sarthaknagoshe2002** [comment]
@afercia Thats completely fine.
I am unsure about why we need or do not need separation but I feel that we should either add the label or remove the separation if separation not needed.
Since, you have suggested to remove the separation I feel lets go your way.
Also, should I continue working on this issue or should we wait for the design feedbacks.

**Comment by afercia** [comment]
I would suggest to wait a couple days for more design feedback, if needed. Personally I'd lean towards removing the separation.
On a procedural note: design feedback can be provided by any contributor, not just by the contributors who belong to the @WordPress/gutenberg-design group, thanks.

**Comment by sarthaknagoshe2002** [comment]
@afercia Do you mind providing insights on what exactly needs design feedbacks to close this issue.
I would like to suggest design if possible.

**Comment by afercia** [comment]
As @jameskoster mentioned, It would be good to have the design team have a look at removing the separation between the groups, which seems the most sensible option.
